src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
-    float initial_speed;
+    float initial_linear_speed;
     // The final speed at the end of the trajectory [m/s]
-    float final_speed;
-} FirmwareRobotPathParameters_t;
+    float final_linear_speed;
 
-// Struct that defines a single point on a trajectory
-// Includes the Position and Time data corresponding to that point
-typedef struct PositionTrajectoryElement
-{
-    Vector2d_t position;
-    float time;
-} PositionTrajectoryElement_t;
+
+} FirmwareRobotPathParameters_t;
 
 typedef struct PositionTrajectory
 {
-    PositionTrajectoryElement_t* trajectory_elements;
-    FirmwareRobotPathParameters_t path_parameters;
-    float* speed_profile;
+    float x_position[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float y_position[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float orientation[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float linear_speed[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float angular_speed[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float time_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
 } PositionTrajectory_t;
 
-// Struct that defines a single point on a velocity trajectory
-// Includes the velocity and Time data corresponding to that point
-typedef struct VelocityTrajectoryElement
-{
-    Vector2d_t linear_velocity;
-    float angular_velocity;
-    float time;
-} VelocityTrajectoryElement_t;
-
 typedef struct VelocityTrajectory
 {
-    VelocityTrajectoryElement_t* trajectory_elements;
-    FirmwareRobotPathParameters_t path_parameters;
+    float x_velocity[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float y_velocity[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float angular_velocity[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float time_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
 } VelocityTrajectory_t;
 
 typedef enum TrajectoryPlannerGenerationStatus
 {
     OK,
     FINAL_VELOCITY_TOO_HIGH,
     INITIAL_VELOCITY_TOO_HIGH,
-
     // INTERPOLATION_ELEMENT_MAXED_OUT is returned when the constant time interpolation
-    // function uses up all the avaiable array space provided to it. This can happen
+    // function uses up all the available array space provided to it. This can happen
     // because constant interpolation period trajectories do not necessarily have the same
-    // number of elements as their constant arc-length counterparts
+    // number of elements as their constant parameterization counterparts
     INTERPOLATION_ELEMENT_MAXED_OUT,
 } TrajectoryPlannerGenerationStatus_t;
 
 /**
- * Returns a planned trajectory with the list of guarantees based on the assumptions below
+ * Function generates the segment lengths and the states at each node along the specified
+ * constant-parameterization segments
+ *
+ * @pre All arrays must be pre-allocated up to TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param t_start [in] The starting parameterization value
+ *
+ * @param t_end  [in] The ending parameterization value
+ *
+ * @param poly  [in] The 1d polynomial representing the path along the trajectory
```

Comment:
perhaps `path_1d` might be a more descriptive name?

Reply:
Changed